### ISSUE TITLE
Implement Momentum without using built-in optimizer

### ISSUE THREAD

**Issue by thushv89** [thushv89/ConvNets#10]

**Comment by thushv89** [comment]
![image](https://cloud.githubusercontent.com/assets/1381369/23181248/60e298fc-f8c8-11e6-9475-bfc637f93109.png)

Here are the loss and accuracy comparisons for SGD and Momentum
We can see that momentum probably overfits the training set 
As the training set is not uniform, this lead to poor test error
